### PR TITLE
Viite 398  Initial load of Road address history 19.11.2017 to devtest

### PR DESCRIPTION
**Pull request by plourenco90**

### CODE REVIEW

File: digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/util/DataFixture.scala [hunk 1]
```diff
@@ -92,6 +92,14 @@ object DataFixture {
     println()
   }
 
+  def importRoadAddressesHistory(isDevDatabase: Boolean): Unit = {
```
Comment (oappicgi) [inline digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/util/DataFixture.scala]: i dont think isdevdatabase boolean is needed

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala [hunk 2]
```diff
@@ -328,6 +423,28 @@ class AssetDataImporter {
...
 
+  def importRoadAddressHistory(conversionDatabase: DatabaseDef, importDate: String): Unit = {
+
+    print(s"\n${DateTime.now()} - ")
+    println("Starting to import history data %s ".format(importDate))
+
+    val roadMaintainerElys = Seq(0, 1, 2, 3, 4, 8, 9, 10, 12, 14)
+
+    //Delete all the previous road history
+    OracleDatabase.withDynTransaction{
```
Comment (oappicgi) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala]: should we drop all the roadaddress and lrm table content?

File: digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/util/DataFixture.scala (hunk 1 above)
Comment (plourenco90) [reply digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/util/DataFixture.scala]: yes, you are right!

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala (hunk 2 above)
Comment (plourenco90) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala]: well, for now it is just deleting the history with "END_DATE" not null, we could delete all but we would lose the "current road" and would have to reinsert it

Comment (oappicgi) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala]: yea, thought we wanted it like that which is why i asked it on daily to be sure. Didn't run test, but code seems ok to me.

**Comment by plourenco90** [comment]
It now contains VIITE-936 and VIITE-937

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala [hunk 3]
```diff
@@ -277,6 +279,129 @@ class AssetDataImporter {
...
+        (address._7, address._8, SideCode.TowardsDigitizing.value)
+      } else {
+        (address._8, address._7, SideCode.AgainstDigitizing.value)
+      }
+      val (x1, y1, x2, y2) = if (sideCode == SideCode.TowardsDigitizing.value)
+        (address._13, address._14, address._15, address._16)
+      else
+        (address._15, address._16, address._13, address._14)
+
+      print(s"insert into ROAD_ADDRESS (id, lrm_position_id, road_number, road_part_number, track_code, discontinuity, START_ADDR_M, END_ADDR_M, start_date, end_date, created_by, VALID_FROM, geometry, floating, road_type, ely, terminated) values " +
```
Comment (tvenhola) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/util/AssetDataImporter.scala]: debug print

File: digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/dao/RoadAddressDAOSpec.scala [hunk 4]
```diff
@@ -30,6 +32,46 @@ class RoadAddressDAOSpec extends FunSuite with Matchers {
     }
   }
 
+  test("insert roadaddress duplicate info check") {
```
Comment (tvenhola) [inline digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/dao/RoadAddressDAOSpec.scala]: Would be better to use intercept to catch the exception and to check that error message

Comment (plourenco90) [reply digiroad2-viite/src/test/scala/fi/liikennevirasto/viite/dao/RoadAddressDAOSpec.scala]: You are right, its done

**Pull request merged**